core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1210,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (newName != null && !newName.trim().equals(name.trim())) {
```

Comment:
@csimons you should compare whole line 
`newName != null && !newName.trim().equals(name.trim())` with 
`!StringUtils.equals(trim(name), trim(newName)) ` which actually shorter, even in form
`!StringUtils.equals(trimToEmpty(name), trim(newName))`